src/ProjectSystem/Environment.fs

```diff
@@ -161,10 +162,96 @@ module Environment =
   /// because 3.x is the minimum runtime that we support for FSI, we want to float to the latest
   /// 3.x runtime that the user has installed, to prevent hard-coding.
   let latest3xRuntimeVersion dotnetRoot =
-    let minRuntimeVersion = FSIRefs.NugetVersion(3,0,0,"")
+    let minRuntimeVersion = NugetVersion.NugetVersion(3,0,0,"")
     lazy (
       match FSIRefs.runtimeVersions dotnetRoot with
       | None -> None
       | Some sortedRuntimeVersions ->
         maxVersionWithThreshold minRuntimeVersion sortedRuntimeVersions
     )
+
+  /// attempts to read the specified SDK version from the global.json file in the workspace root.
+  let versionFromGlobalJson (globalJsonPath: string): NugetVersion.NugetVersion option =
```

Comment:
note to future parties: I don't think we should parse global.json ourselves anymore: instead shell execute `dotnet --version` from the working directory of the workspace in question. this frees us from having to implement rollforward support.